src/EnergyPlus/HeatBalanceManager.cc

```diff
@@ -5173,6 +5175,89 @@ namespace HeatBalanceManager {
+                                                                         Loop,
+                                                                         state.dataIPShortCut->cAlphaArgs,
+                                                                         NumAlpha,
+                                                                         state.dataIPShortCut->rNumericArgs,
+                                                                         NumNumeric,
+                                                                         IOStat,
+                                                                         state.dataIPShortCut->lNumericFieldBlanks,
+                                                                         state.dataIPShortCut->lAlphaFieldBlanks,
+                                                                         state.dataIPShortCut->cAlphaFieldNames,
+                                                                         state.dataIPShortCut->cNumericFieldNames);
+                if (UtilityRoutines::IsNameEmpty(state, state.dataIPShortCut->cAlphaArgs(1), cCurrentModuleObject, ErrorsFound)) {
+                    ShowContinueError(
+                        state, "...each SurfaceProperty:SolarIncidentInside name must not duplicate other SurfaceProperty:SolarIncidentInside name");
+                    continue;
+                }
+
+                // Assign surface number
+                int SurfNum = UtilityRoutines::FindItemInList(state.dataIPShortCut->cAlphaArgs(1), state.dataSurface->Surface);
+                if (SurfNum == 0) {
+                    ShowSevereError(state,
+                                    std::string{RoutineName} + cCurrentModuleObject + "=\"" + state.dataIPShortCut->cAlphaArgs(1) +
+                                        ", object. Illegal value for " + state.dataIPShortCut->cAlphaFieldNames(1) + " has been found.");
+                    ShowContinueError(state,
+                                      state.dataIPShortCut->cAlphaFieldNames(1) + " entered value = \"" + state.dataIPShortCut->cAlphaArgs(1) +
+                                          "\" no corresponding surface (ref BuildingSurface:Detailed) has been found in the input file.");
+                    ErrorsFound = true;
+                } else {
+                    if (state.dataSurface->Surface(SurfNum).Class == DataSurfaces::SurfaceClass::Window &&
+                        state.dataSurface->Surface(SurfNum).ExtBoundCond == DataSurfaces::ExternalEnvironment) {
+                        int ConstrNum = state.dataSurface->Surface(SurfNum).Construction;
+                        int MaterNum =
+                            state.dataConstruction->Construct(ConstrNum).LayerPoint(state.dataConstruction->Construct(ConstrNum).TotLayers);
+                        bool withNoncompatibleShades =
+                            (state.dataMaterial->Material(MaterNum).Group == DataHeatBalance::MaterialGroup::Shade ||
```

Comment:
Do you know how to use `BITF_TEST_ANY`? 📌 Someone convert that macro into a `constexpr inline` function please.

Reply:
Oh! I didn't know that. I'll use BITF_TEST_ANY instead

Reply:
These are not bit fields, and the integer enum choices are not powers of 2, so I assume we can't use BITF_TEST_ANY? Would a switch case be clearer?

Reply:
I know they are not bitfields. But look at how `BITF_TEST_ANY` is used (it's already used in a few places). Because enumerations are dense literal integers, they can be temporarily converted to bitfields for comparison purposes. That way you can compare the `MaterialGroup` variable to all the ones you are interested in one operation (instead of however many you are using here). Because the conversion of a literal to a bitfield is itself a literal operation, the compiler can collapse all of those into a single literal at compile time (like a `constexpr`). You do not want to do this as a `switch` because, in general, you do not want to convert a data-only operation into one that also involves control flow.

Reply:
Is the goal here to improve readability or do we think the `BITF_TEST_ANY` compiles to something faster? Is the compiler not able to optimize the existing approach?

Reply:
Would the compiler be able to figure out some even smarter checks? Like maybe in some special case, the compiler doesn't need to do bit shifting of `state.dataMaterial->Material(MaterNum).Group`, but can check a single bit? Or in less special cases, would it automatically do what the BITF approach would do?

Reply:
Yes, the `BITF` approach is faster. You are right that the compiler "should" be able to figure this out because all comparisons are literal integers and the largest one is smaller than 64 (you cannot make integers larger than 64 into bitfields because integers are 64 bits), but the fact is that they don't do it(at least `g++ -O3` doesn't, maybe there are some specific optimization flags that would do this but I cannot find them). 

Also, compilers don't optimize for special cases (and they definitely shouldn't do it in this case) because you need a data-dependent branch to check whether the special case is in effect and mis-predicting that branch at runtime will cost you more than the general optimization will help you. TL;DR: this optimization is not something that compilers do right now. 

As far as readability, I agree that `BITF` is less readable than it could be right now. That's because it's implemented as a single macro which takes two arguments. It may be slightly more readable if it were implemented as a group of macros `BITF_TEST_ANY3`, `BITF_TEST_ANY4`, etc. because then the macros could do the `BITF` operations internally rather than you having to do them in the arguments. Also, with `constexpr inline` functions there is no real reason to have macros anymore.

Reply:
Actually, I stand corrected! Maybe I was looking at the wrong disassembly file at first but `g++ -O3` does do this optimization (in fact, it even does it with `-O2`)! Okay, there is no need to do this and we can probably stop doing this elsewhere too.

Reply:
Thanks for checking out the compiler behavior. Glad the compilers are smart enough to handle it.

Reply:
Smarter than me apparently! 😭

Reply:
Seriously, I am always up for learning a better way of doing something. You will never see anyone change their mind faster about something than I do when I am shown (or find) a better way. Thanks for asking the question that made me take another look at this!

Reply:
Thanks for being so patient in explaining everything. Very helpful for noobs like me to get to know all the millions of ways to do things in c++ lol

Reply:
I've learned quite a bit about C++ myself over the past several years. I programmed in it in the mid-1990's and then didn't really look at it again until the "EnergyPlus 10X" project four years ago. In the meantime, I only programmed in C and python with some Object Pascal, Java, and a bunch of assembly language. C++ and its libraries are much different than what I remembered with many new features, mostly good (some like `auto` bad), and I have been trying to learn them too. Just a few weeks ago I learned that c++17 requires RVO (return value optimization) in optimized compilation which changes what I thought about returning a `struct` from a function (I thought it was a "no-no" for performance reasons, but it's actually mostly fine now). The week before that I learned what a "transparent" template function is. Always new stuff to learn.